Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -2,24 +2,38 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <thread>
+
 #include <QMouseEvent>
 #include <QRegExp>
 #include <QString>
-#include <thread>
 
 #include "DolphinQt2/Config/Mapping/MappingButton.h"
 
 #include "Common/Thread.h"
+#include "DolphinQt2/Config/Mapping/IOWindow.h"
+#include "DolphinQt2/Config/Mapping/MappingCommon.h"
 #include "DolphinQt2/Config/Mapping/MappingWidget.h"
 #include "DolphinQt2/Config/Mapping/MappingWindow.h"
 #include "InputCommon/ControlReference/ControlReference.h"
 #include "InputCommon/ControllerEmu/ControllerEmu.h"
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/ControllerInterface/Device.h"
+
+static QString TrunicateDisplayString(const QString& string)
+{
```

Comment:
Hm, it might be worth the time investment if it were used elsewhere, but I don't think it really is.